### PR TITLE
Add .NET 5.0 support to Akka.Http.Shim

### LINKED ISSUE #23: Akka.Http.Shim uses Microsoft.AspNetCore 2.2.0, which is not compatible with .NET 5.0
As of right now, we're still aiming for .NET Standard 2.0 support, hence why we're using Microsoft.AspNetCore 2.2.0, but it seems like there is no update planned for this package in the future. Including Akka.Http.Shim in a project targetting .NET 5.0 will compile just fine but will fail to run as it will be missing some transient dependencies.

To support .NET 5.0 and up, we would need to do a dual deploy for both .NET Standard 2.0 and .NET 5.0, problem is that they changed the API in the library, so we would need to do conditional coding, and we will have to test the shim in both framework.

### PR DESCRIPTION
**Pull request by Arkatufus**
Closes #23

There are some build script changes, need to make sure that PR validations are correct

### CODE REVIEW

**Comment by Arkatufus** [comment]
Adding .NET FX 4.7.1 causes the spec to become racy

File: build-system/pr-validation.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
     parameters:
       name: 'linux_pr'
       displayName: 'Linux PR Validation'
-      vmImage: 'ubuntu-16.04'
+      vmImage: 'ubuntu-18.04'
```
Comment (Aaronontheweb) [inline build-system/pr-validation.yaml:30]: LGTM

File: src/cluster.bootstrap/Akka.Management.Cluster.Bootstrap/Akka.Management.Cluster.Bootstrap.csproj [hunk 2]
```diff
@@ -1,7 +1,10 @@
 <Project Sdk="Microsoft.NET.Sdk">
 
     <PropertyGroup>
-        <TargetFramework>$(LibraryFramework)</TargetFramework>
+        <TargetFrameworks>$(LibraryFramework);$(NetFramework)</TargetFrameworks>
```
Comment (Aaronontheweb) [inline src/cluster.bootstrap/Akka.Management.Cluster.Bootstrap/Akka.Management.Cluster.Bootstrap.csproj:4]: Do we have any `#if`s or separate sets of dependencies for these packages?

File: src/management/Akka.Http.Shim/Akka.Http.Shim.csproj [hunk 3]
```diff
@@ -16,7 +18,8 @@
 
   <ItemGroup>
     <PackageReference Include="Akka" Version="$(AkkaVersion)" />
-    <PackageReference Include="Microsoft.AspNetCore" Version="2.2.0" />
+    <PackageReference Condition="'$(TargetFramework)' == '$(LibraryFramework)'" Include="Microsoft.AspNetCore" Version="2.2.0" />
+    <FrameworkReference Condition="'$(TargetFramework)' == '$(NetFramework)'" Include="Microsoft.AspNetCore.App" />
```
Comment (Aaronontheweb) [inline src/management/Akka.Http.Shim/Akka.Http.Shim.csproj:22]: There we go - that answers my question from earlier.

File: src/management/Akka.Http.Shim/Dsl/Model/HttpMessage.cs [hunk 4]
```diff
@@ -66,7 +66,11 @@ private HttpRequest(Microsoft.AspNetCore.Http.HttpRequest request)
         {
             _request = request;
             var input = new byte[Convert.ToInt32(request.ContentLength)];
+#if NET5_0
+            request.Body.ReadAsync(input, 0, input.Length).Wait();
```
Comment (Aaronontheweb) [inline src/management/Akka.Http.Shim/Dsl/Model/HttpMessage.cs:70]: Ugly but this data should all be local to the machine by the time this call is made anyway.

**Review by Aaronontheweb — APPROVED** [review]
LGTM

File: src/management/Akka.Http.Shim/Dsl/Model/HttpMessage.cs (hunk 4 above)
Comment (Arkatufus) [reply src/management/Akka.Http.Shim/Dsl/Model/HttpMessage.cs:70]: Yes, there's no sync version for the new ASP.NET, everything are async

Comment (Aaronontheweb) [reply src/management/Akka.Http.Shim/Dsl/Model/HttpMessage.cs:70]: I figured as much

**Pull request merged**